content/kubevirt/storage.md

```diff
@@ -0,0 +1,182 @@
+
+    In vSphere plain storage consumption, the storage capacity is provided using large volumes (aka described as LUNs) as datastores and then the vmdk "block" devices carved out as files coming from vmfs residing on the datastores. A similar approach is used by RHV.
+
+    In OCP Virtualization instead, every persistent volume for a virtual disk coming from a block storage must be provided as a real world block storage device. This usually leads to trouble on the OCP nodes as well as on the enterprise storage: the number of PVs used for VMs define the number of block volumes to create on the enterprise storage subsystems, to be handled in SAN mappings, and to be managed by FC/iSCSI and multi-pathing on the OCP nodes.
+
+## Overview of the CSI Components
+
+The diagram below provides a high-level overview of the CSI components in an OpenShift cluster. The CSI driver coordinates the creation and management of the volume on the storage device, mounting of the volume to the node hosting the virtual machine, and implementing features like snapshots and volume resizing at the storage volume level.
+
+* *Source: [Docs Red Hat - OCP 4.17 Using CSI](https://docs.openshift.com/container-platform/4.17/storage/container_storage_interface/persistent-storage-csi.html)*
+
+![CSI Storage](storage_csi.png)
+
+## CSI drivers supported by OpenShift Container Platform
+
+OpenShift Container Platform installs certain CSI drivers by default, giving users storage options that are not possible with in-tree volume plugins.
+
+The following table describes the CSI drivers that are installed with OpenShift Container Platform supported by OpenShift Container Platform and which CSI features they support, such as volume snapshots and resize.
+
+### *Applies to OCP-V version 4.17*
+
+* *Source: [Docs Red Hat - OCP 4.17 Using CSI](https://docs.openshift.com/container-platform/4.17/storage/container_storage_interface/persistent-storage-csi.html)*
+
+![CSI Driver Support](csi-driver-support.png)
+
+* (1) Requires vSphere version 7.0 Update 3 or later for both vCenter Server and ESXi.
+  * Does not support fileshare volumes.
+
+* (2) Offline volume expansion: minimum required vSphere version is 6.7 Update 3 P06
+  * Online volume expansion: minimum required vSphere version is 7.0 Update 2.
+
+* (3) Does not support offline snapshots or resize. Volume must be attached to a running pod.
+
+* (4) Azure File cloning does not supports NFS protocol. It supports the azurefile-csi storage class, which uses SMB protocol.
+  * Azure File cloning and snapshot are Technology Preview features.
+
+## Configuring local storage for virtual machines
+
+Official documentation: [Configuring local storage for virtual machines
+](https://docs.openshift.com/container-platform/latest/cnv/cnv_virtual_machines/cnv_virtual_disks/cnv-configuring-local-storage-for-vms.html)
+
+## Create a backing directory on each node
```

Comment:
Please fix the indenting. It should be

* Configuring local storage for virtual machines
  * Create a backing directory on each node
    *  Create MachineConfig

As it was in the orgiginal one.

Reply:
![Screenshot 2025-01-10 at 16 21 19](https://github.com/user-attachments/assets/278e55bb-83f5-4870-9699-a97a69baef8b)

Reply:
Ok the original was not perfect as well:
![image](https://github.com/user-attachments/assets/a6eb02bd-c83e-434a-83e9-b194002a08c2)

Reply:
Done